files/check_hddtemp

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Author: Michel Weitbrecht (michel.weitbrecht@stuvus.uni-stuttgart.de)
+
+import os
+import subprocess
+import argparse
+import json
+from common import doCheck
+
+POSSIBLE_STATES = ('OK', 'WARNING', 'CRITICAL', 'UNKNOWN')
+
+def raiseState(newState, ret):
+    if newState not in POSSIBLE_STATES:
+        ret['state'] = 'UNKNOWN'
+    elif POSSIBLE_STATES.index(newState) > POSSIBLE_STATES.index(ret['state']):
+        ret['state'] = newState
+
+
+def check(ret):
+    ret['name'] = 'hddtemp'
+
+    parser = argparse.ArgumentParser(
+        prog='check_hddtemp',
+        description='Check the temperature of all SCSI-like devices.')
+    parser.add_argument('-tw', '--temp-warn',
+                        help="Return WARNING if the temperature in degree Celcius of any device is higher than this", type=int,
+                        default=40)
+    parser.add_argument('-tc', '--temp-crit',
+                        help="Return CRITICAL if the temperature in degree Celcius of any device is higher than this", type=int,
+                        default=50)
+    parser.add_argument('-i', '--ignore',
+                        help="Ignore this device, e.g. 'sda'. May be provided multiple times.", default=[], action='append')
+    args = vars(parser.parse_args())
+
+    lsblk_process = subprocess.Popen(['lsblk', '-SJ', '--output=NAME,MODEL'], stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    lsblk_json_output = lsblk_process.communicate()[0].decode('ISO-8859-1')
```

Comment:
Are you sure that's not utf-8?

Reply:
Yes, I tried that first, but it doesn't work for all devices, e.g. my local SSD is not in the database and outputs characters not included in `UTF-8`.